mlir/lib/Dialect/MemRef/IR/MemRefOps.cpp

```diff
@@ -3259,6 +3267,15 @@ OpFoldResult TransposeOp::fold(FoldAdaptor) {
   return {};
 }
 
+bool TransposeOp::isIdentity() {
+  ArrayRef<int64_t> permutationArray = getPermutation();
+  for (int64_t i = 0, rank = permutationArray.size(); i < rank; ++i)
```

Comment:
Nit: the norm is to use iterator utils like this

```
for (auto [index, dim] : llvm::enumerate(getPermutation()))
  if (index != dim)
    return false;
return true;
```

Reply:
Thank you!